src/waveform/renderers/allshader/waveformrendererlrrgb.cpp

```diff
@@ -193,15 +205,15 @@ void WaveformRendererLRRGB::paintGL() {
             m_vertices.addRectangle(fpos - 0.5f,
                     halfBreadth,
                     fpos + 0.5f,
-                    halfBreadth + heightFactor[chn] * std::sqrt(maxAll));
+                    halfBreadth + heightFactor[chn] * maxAll);
             m_colors.addForRectangle(red, green, blue);
         }
 
         xVisualSampleIndex += visualIncrementPerPixel;
     }
 
-    DEBUG_ASSERT(linesReserved == m_vertices.size());
-    DEBUG_ASSERT(colorsReserved == m_colors.size());
+    DEBUG_ASSERT(reserved == m_vertices.size());
+    DEBUG_ASSERT(reserved == m_colors.size());
```

Comment:
so I guess m_colors and m_vertices are the same size? (do we need to check both these things? This seems like an odd place to enforce this restriction rather than when reserved is calculated)

Reply:
yes, they are the same size. we could remove these asserts, it's just to make sure i didn't make a mistake when filling these vectors. but they don't enforce anything, they merly check. and checking doesn't hurt.

Reply:
Got it.  Wouldn't it be better to check them back up where reserved is created, then?  If there's something wrong with that size, then we will encounter problems before we get to this check.

Reply:
What is being checked with this assert is that the amount of data that we push to the m_vertices / m_colors vectors is correct (in m_vertices.addRectangle and m_colors.addForRectangle), to make sure we reserve exactly the amount we are going to push.